flake8_trio.py

```diff
@@ -63,32 +75,21 @@ def __eq__(self, other: Any) -> bool:
         )
 
 
-HasLineInfo = Union[ast.expr, ast.stmt, ast.arg, ast.excepthandler, Statement]
-
-
-class TrioScope:
-    def __init__(self, node: ast.Call, funcname: str):
-        self.node = node
-        self.funcname = funcname
-        self.variable_name: Optional[str] = None
-        self.shielded: bool = False
-        self.has_timeout: bool = True
+HasLineCol = Union[ast.expr, ast.stmt, ast.arg, ast.excepthandler, Statement]
 
-        # scope.shield is assigned to in visit_Assign
 
-        if self.funcname == "CancelScope":
-            self.has_timeout = False
-            for kw in node.keywords:
-                # Only accepts constant values
-                if kw.arg == "shield" and isinstance(kw.value, ast.Constant):
-                    self.shielded = kw.value.value
-                # sets to True even if timeout is explicitly set to inf
-                if kw.arg == "deadline":
-                    self.has_timeout = True
-
-    def __str__(self):
-        # Not supporting other ways of importing trio, per TRIO106
-        return f"trio.{self.funcname}"
+def get_matching_call(
+    node: ast.AST, *names: str, base: str = "trio"
```

Comment:
A bit obscuring to default to `"trio"` when no longer having it in the function name, but probs fine.